sdk/python/v1beta1/kubeflow/katib/api/katib_client_test.py

```diff
@@ -0,0 +1,286 @@
+    experiment_namespace = "test"
+
+    if name is not None:
+        metadata = V1ObjectMeta(name=name, namespace=experiment_namespace)
+    elif generate_name is not None:
+        metadata = V1ObjectMeta(generate_name=generate_name, namespace=experiment_namespace)
+    else:
+        metadata = V1ObjectMeta(namespace=experiment_namespace)
+
+    algorithm_spec=V1beta1AlgorithmSpec(
+        algorithm_name="random"
+    )
+
+    objective_spec=V1beta1ObjectiveSpec(
+        type="minimize",
+        goal= 0.001,
+        objective_metric_name="loss",
+    )
+
+    parameters=[
+        V1beta1ParameterSpec(
+            name="lr",
+            parameter_type="double",
+            feasible_space=V1beta1FeasibleSpace(
+                min="0.01",
+                max="0.06"
+            ),
+        ),
+        V1beta1ParameterSpec(
+            name="momentum",
+            parameter_type="double",
+            feasible_space=V1beta1FeasibleSpace(
+                min="0.5",
+                max="0.9"
+            ),
+        ),
+    ]
+
+    trial_template = generate_trial_template()
+
+    experiment = generate_experiment(
+        metadata,
+        algorithm_spec,
+        objective_spec,
+        parameters,
+        trial_template
+    )
+    return experiment
+
+
+test_create_experiment_data = [
+    (
+        "experiment name and generate_name missing",
+        {"experiment": create_experiment()},
+        ValueError,
+    ),
+    (
+        "create_namespaced_custom_object timeout error",
+        {
+            "experiment": create_experiment(name="experiment-mnist-ci-test"),
+            "namespace": "timeout",
+        },
+        TimeoutError,
+    ),
+    (
+        "create_namespaced_custom_object conflict error",
+        {
+            "experiment": create_experiment(name="experiment-mnist-ci-test"),
+            "namespace": "conflict",
+        },
+        Exception,
+    ),
+    (
+        "create_namespaced_custom_object runtime error",
+        {
+            "experiment": create_experiment(name="experiment-mnist-ci-test"),
+            "namespace": "runtime",
+        },
+        RuntimeError,
+    ),
+    (
+        "valid flow with experiment type V1beta1Experiment and name",
+        {
+            "experiment": create_experiment(name="experiment-mnist-ci-test"),
+            "namespace": "test",
+        },
+        "success",
+    ),
+    (
+        "valid flow with experiment type V1beta1Experiment and generate_name",
+        {
+            "experiment": create_experiment(generate_name="experiment-mnist-ci-test"),
+            "namespace": "test",
+        },
+        "success",
+    ),
+    (
+        "valid flow with experiment type V1beta1Experiment and name and generate_name",
+        {
+            "experiment": create_experiment(
+                name="experiment-mnist-ci-test",
+                generate_name="experiment-mnist-ci-test",
+            ),
+            "namespace": "test"
```

Comment:
If we pass generate_name and name how does Kubernetes Python client create the Experiment ?

Reply:
I suppose that `name` is used when both `name` and `generate_name` are provided.
Ref: https://github.com/tariq-hasan/katib/blob/master/sdk/python/v1beta1/kubeflow/katib/api/katib_client.py#L102-L105.

Would you suggest making any changes to this case?

Reply:
These parameters are mutually exclusive. Users can set `name` or `generate_name`

Reply:
Okay. I have removed the two test cases where both `name` and `generate_name` are specified.